charts/actions-runner-controller/templates/deployment.yaml

```diff
@@ -6,6 +6,7 @@ metadata:
   labels:
     {{- include "actions-runner-controller.labels" . | nindent 4 }}
 spec:
+  replicas: {{ .Values.replicaCount }}
```

Comment:
This should be set when and only when replicaCount is defined. Otherwise, HRA is unable to auto-scale the replicas value. I'll add a commit myself to make the change.